tests/unistd/links.out

```diff
@@ -21,11 +21,11 @@ symlink/normal
 ret: 0
 errno: 0
 readlink(created link)
-ret: 20
+ret: 29
 errno: 0
-result: new-nonexistent-path
+result: /working/new-nonexistent-path
 
 readlink(short buffer)
 ret: 3
 errno: 0
-result: ../-nonexistent-path
+result: /thrking/new-nonexistent-path
```

Comment:
@kripken FS.Readlink returns `/there!`, in library.js this is then shortened to 3 characters (`/th`) because the 4 byte buffer and returned. However, `buffer` still contains the previous value (which is /working/new-nonexistent-path) and thus just overwrites the first bytes. That's why.

Previously the result used to be new-nonexistent-path, so that's why it changed.